app/channels/discussion_post_list_channel.rb

```diff
@@ -0,0 +1,14 @@
+class DiscussionPostListChannel < ApplicationCable::Channel
+  def self.notify!(discussion, iteration)
+    broadcast_to("#{discussion.uuid}_#{iteration.idx}", {})
+  end
+
+  def subscribed
+    discussion = Solution::MentorDiscussion.find_by!(uuid: params[:discussion_id])
+    iteration = discussion.solution.iterations.find_by!(idx: params[:iteration_idx])
+
+    return unless discussion.viewable_by?(current_user)
```

Comment:
I think re-ordering these two lines would save a query when the discussion is not viewable?

```suggestion
    return unless discussion.viewable_by?(current_user)

    iteration = discussion.solution.iterations.find_by!(idx: params[:iteration_idx])    
```

Reply:
Nice! Thanks :)